source/coap_connection_handler.c

```diff
@@ -793,9 +801,13 @@ coap_conn_handler_t *connection_handler_create(receive_from_socket_cb *recv_from
 
     return handler;
 }
-void connection_handler_destroy(coap_conn_handler_t *handler)
+
+void connection_handler_destroy(coap_conn_handler_t *handler, bool multicast_group_leave)
 {
     if(handler){
+        if (multicast_group_leave) {
```

Comment:
Leaving is not required. Closing the socket is sufficient. (Which might render a lot of the changes redundant?)